### DISCUSSION TITLE
Fine tuning trouble

### DISCUSSION THREAD

**Discussion by jdavidpoling** [WildlifeDatasets/wildlife-datasets#8] [General]
Hi. I am having trouble finding resources on how to fine tune a model on my own dataset. Is more information available on how to do this? Sorry if I missed something already in the repo.

**Comment by sadda** [comment]
Hi, wildlife-datasets is only about datasets, not about models. However, based on the datasets here, we trained a model [MegaDescriptor](https://huggingface.co/BVRA/MegaDescriptor-L-384). Either use it or some other model, such as [MiewID](https://huggingface.co/conservationxlabs/miewid-msv3), designed for animal re-identification. Then apply the standard fine-tuning; you may want to take a look at the [hyperparameters](https://github.com/WildlifeDatasets/wildlife-tools/blob/main/baselines/training/MegaDescriptor-L-384.ipynb), which we used for training. Please let me know if you get any interesting results. Thanks a lot :)

Lukas

**Reply by jdavidpoling** [reply]
Thank you! I have found what I need and successfully gotten going. I'll reach back out if there is anything notable. :)

**Reply by sadda** [reply]
Perfect, thanks a lot.

**Comment by ahrendsen** [comment]
I'm also in the process of fine tuning, and wanted to ask some questions to set expectations. For background, I'm trying to ID newts and salamanders. I've identified that spectral reflections on the surface of the wet skin of the animals can present a significant difficulty for computer vision algorithms because they look like notable features, when they are in fact noise.

![IMG_0315](https://github.com/user-attachments/assets/af3767e7-da31-4aa3-b296-a05dc0332042)

My initial attempt to ID my database had an accuracy of 58% as reported by the example script in the root of wildlife-tools. My question is, what amount of accuracy improvement might you expect from training the model specifically on these photos? I have 17,000 photos, most of them with individuals IDed previously using I3S. Would you start from scratch with the swin backbone, or would you build off of the MegaDescriptor model?

**Reply by sadda** [reply]
I would not train MegaDescriptor from scratch; I would use one or two epochs to finetune. This year, we organised the [AnimalCLEF competition](https://www.kaggle.com/competitions/animal-clef-2025), where one of the animals were fire salamanders, see [our paper](https://ceur-ws.org/Vol-4038/paper_231.pdf). The task was open-set, which means that, in addition to recognising known individuals, some individuals were new and needed to be recognised as such. MegaDescriptor achieved around 30%, WildFusion was at 45% and the winning submission was over 70%. The [winners](https://ceur-ws.org/Vol-4038/paper_251.pdf) also published their [code online](https://github.com/XXXM1R0XXX/AnimalCLEF2025), which may be worth checking. I still need to go through it properly.
